contracts/rbn-staking/IncentivizedVotingLockup.sol

```diff
@@ -497,26 +577,45 @@ contract IncentivisedVotingLockup is
    */
   function _withdraw(address _addr) internal nonReentrant {
     LockedBalance memory oldLock =
-      LockedBalance({end: locked[_addr].end, amount: locked[_addr].amount});
+      LockedBalance({
+        end: locked[_addr].end,
+        shares: locked[_addr].shares,
+        amount: locked[_addr].amount
+      });
     require(block.timestamp >= oldLock.end, "The lock didn't expire");
     require(oldLock.amount > 0, "Must have something to withdraw");
 
-    uint256 value = SafeCast.toUint256(oldLock.amount);
+    uint256 shares = SafeCast.toUint256(oldLock.shares);
 
-    LockedBalance memory currentLock = LockedBalance({end: 0, amount: 0});
-    totalLocked -= value;
+    LockedBalance memory currentLock =
+      LockedBalance({end: 0, shares: 0, amount: 0});
     locked[_addr] = currentLock;
 
     // oldLocked can have either expired <= timestamp or zero end
     // currentLock has only 0 end
     // Both can have >= 0 amount
     _checkpoint(_addr, oldLock, currentLock);
```

Comment:
as you said, the weight will still reflect the amount they `deposited`, not how much their deposit is worth now due to potential redeems.

for ex: 

```
alice deposits 20 rbn
bob deposits 10 rbn
seize 10 rbn
john deposits 7 rbn
```

the weight for alice will be based on **20 rbn**, not **13.3 rbn**. 
the weight for bob will be based on **10 rbn**, not **6.6 rbn**. 
the weight for john will be based on **7 rbn**